File: src/pytra/runtime/std/math_impl.py
from __future__ import annotations

import pytra.std.math_impl as _m

pi: float = _m.pi
e: float = _m.e


def sqrt(x: float) -> float:
    return _m.sqrt(x)


def sin(x: float) -> float:
    return _m.sin(x)


def cos(x: float) -> float:
    return _m.cos(x)


def tan(x: float) -> float:
    return _m.tan(x)


def exp(x: float) -> float:
    return _m.exp(x)


def log(x: float) -> float:
    return _m.log(x)


def log10(x: float) -> float:
    return _m.log10(x)


def fabs(x: float) -> float:
    return _m.fabs(x)


def floor(x: float) -> float:
    return _m.floor(x)


def ceil(x: float) -> float:
    return _m.ceil(x)


def pow(x: float, y: float) -> float:
    return _m.pow(x, y)

File: src/runtime2/java/pytra-core/std/math_impl.java
// Java math_impl bridge for pytra.std.math.

final class _m {
    private _m() {
    }

    static final double pi = Math.PI;
    static final double e = Math.E;

    static double sqrt(double x) {
        return Math.sqrt(x);
    }

    static double sin(double x) {
        return Math.sin(x);
    }

    static double cos(double x) {
        return Math.cos(x);
    }

    static double tan(double x) {
        return Math.tan(x);
    }

    static double exp(double x) {
        return Math.exp(x);
    }

    static double log(double x) {
        return Math.log(x);
    }

    static double log10(double x) {
        return Math.log10(x);
    }

    static double fabs(double x) {
        return Math.abs(x);
    }

    static double floor(double x) {
        return Math.floor(x);
    }

    static double ceil(double x) {
        return Math.ceil(x);
    }

    static double pow(double x, double y) {
        return Math.pow(x, y);
    }
}
